packages/sdk-react/src/components/receivables/InvoiceDetails/ExistingInvoiceDetails/components/EditInvoiceDetails.tsx

```diff
@@ -0,0 +1,289 @@
+
+  /** Callback that is called when the invoice is updated */
+  onUpdated: (updatedReceivable: ReceivableResponse) => void;
+
+  /** Callback that is called when the user cancels the editing */
+  onCancel: () => void;
+}
+
+export const EditInvoiceDetails = ({
+  invoice,
+  onCancel,
+  onUpdated,
+}: IEditInvoiceDetails) => {
+  const { i18n } = useLingui();
+  const { root } = useRootElements();
+  const { data: counterpartAddresses } = useCounterpartAddresses(
+    invoice.counterpart_id
+  );
+
+  const counterpartShippingAddress = counterpartAddresses?.find((address) => {
+    return (
+      address.city === invoice.counterpart_shipping_address?.city &&
+      address.country === invoice.counterpart_shipping_address.country &&
+      address.line1 === invoice.counterpart_shipping_address.line1
+    );
+  });
+
+  const methods = useForm<ICreateReceivablesForm>({
+    resolver: yupResolver(getUpdateInvoiceValidationSchema(i18n)),
+    defaultValues: useMemo(
+      () => ({
+        /** Customer section */
+        counterpart_id: invoice.counterpart_id,
+        counterpart_contact: invoice.counterpart_contact,
+        counterpart_vat_id_id: invoice.counterpart_vat_id?.id ?? '',
+        default_shipping_address_id: counterpartShippingAddress?.id ?? '',
+        default_billing_address_id: '',
+
+        /** Entity section */
+        entity_vat_id_id: invoice.entity_vat_id?.id ?? '',
+        fulfillment_date: invoice.fulfillment_date
+          ? new Date(invoice.fulfillment_date)
+          : null,
+        purchase_order: invoice.purchase_order ?? '',
+
+        /** Items section */
+        line_items: invoice.line_items.map((lineItem) => ({
+          quantity: lineItem.quantity,
+          product_id: lineItem.product.id,
+          vat_rate_id: lineItem.product.vat_rate.id,
+          vat_rate_value: lineItem.product.vat_rate.value,
+          name: lineItem.product.name,
+          price: lineItem.product.price,
+          measure_unit_id: lineItem.product.measure_unit_id,
+        })),
+        vat_exemption_rationale: invoice.vat_exemption_rationale ?? '',
+
+        /** Items section */
+        entity_bank_account_id: invoice.entity_bank_account?.id ?? '',
+        payment_terms_id: invoice.payment_terms?.id ?? '',
+      }),
+      [
+        counterpartShippingAddress?.id,
+        invoice.counterpart_contact,
+        invoice.counterpart_id,
+        invoice.counterpart_vat_id?.id,
+        invoice.entity_bank_account?.id,
+        invoice.entity_vat_id?.id,
+        invoice.fulfillment_date,
+        invoice.line_items,
+        invoice.payment_terms?.id,
+        invoice.purchase_order,
+        invoice.vat_exemption_rationale,
+      ]
+    ),
+  });
+
+  const [actualCurrency, setActualCurrency] = useState(invoice.currency);
+
+  const {
+    handleSubmit,
+    setValue,
+    formState: { isDirty },
+  } = methods;
+
+  useEffect(() => {
+    setValue('default_shipping_address_id', counterpartShippingAddress?.id);
```

Comment:
I suppose, this is the reason why the form state is always `touched`. It would be better to avoid using such hacks.

Reply:
Updated fc7432639ab94a4a829676230bb7df6fe0c94fd5